### ISSUE TITLE
Migration failed to apply to shadow database

### ISSUE THREAD

**Issue by Corydeeee** [prisma/prisma#8888]
### Bug description

We are constantly getting an error when trying to apply a migration due to a schema change. The last column we added 'slugName' will aways generate this error when the migration is applied 
`Error: P3006

Migration `20210624184346_added_slug_name` failed to apply cleanly to the shadow database. 
Error code: P3018
Error:
A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 20210624184346_added_slug_name

Database error code: 1146

Database error:
Table 'prisma_migrate_shadow_db_8ffd3d5c-5d54-4052-91f9-c30857576558.applicationconfig' doesn't exist

Please check the query number 1 from the migration file.
`

This migration is never resolved. Even if we ignore this migration and remove this column / property, and add any new property, the same error occurs. All migrations before this step however run normally, it only appears as new migrations are added. 

### How to reproduce

1. Add any property to the schema. 
2. run 'prisma migrate dev' 
3. Get error

### Expected behavior

The migration should run and column to be added to database. 

### Prisma information

Schema
`datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model ApplicationConfig {
  id                  Int             @id @default(autoincrement())
  appId               String          @unique
  androidAppId        String          @default("")
  appName             String
  slugName            String          @default("")
  accessToken         String
  versionCode         Int             @default(1)
  majorVersion        Int             @default(1)
  majoriOSVersion     Int             @default(1)
  majorAndroidVersion Int             @default(1)
  minorVersion        Int             @default(0)
  patchVersion        Int             @default(0)
  publishStatus       Boolean         @default(false)
  buildStatus         Boolean         @default(false)
  merchantAssets      MerchantAsset[]
}

model MerchantAsset {
  id            Int               @id @default(autoincrement())
  key           String
  value         String
  application   ApplicationConfig @relation(fields: [applicationId], references: [id])
  applicationId Int

  @@unique([key, applicationId], name: "MerchantAsset_key_applicationId_unique_constraint")
}
`

Resulting Migration file

`
-- AlterTable
ALTER TABLE `ApplicationConfig` ADD COLUMN     `slugName` VARCHAR(191) NOT NULL DEFAULT '';

`

### Environment & setup

- OS: MacOS 11.5.1 Big Sur
- Database: MySQL 5.7.17
- Node.js version: v16.6.1

### Prisma Version

```
prisma               : 2.28.0
@prisma/client       : 2.26.0
Current platform     : darwin
Query Engine         : query-engine 89facabd0366f63911d089156a7a70125bfbcd27 (at ../../../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli 89facabd0366f63911d089156a7a70125bfbcd27 (at ../../../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core 89facabd0366f63911d089156a7a70125bfbcd27 (at ../../../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt 89facabd0366f63911d089156a7a70125bfbcd27 (at ../../../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : 89facabd0366f63911d089156a7a70125bfbcd27
Studio               : 0.417.0

```

**Comment by tomhoule** [comment]
Hi @Corydeeee , are you using a native OSX build of MySQL or a docker container/VM/hosted instance?

**Comment by Corydeeee** [comment]
@tomhoule a docker container

**Comment by Corydeeee** [comment]
Hey prisma team, I found a way to avoid this issue today. The problem comes from using Capitalization on the mysql command for some reason. So for some reason if I ran the command
`
ALTER TABLE `ApplicationConfig` ADD COLUMN     `slugName` VARCHAR(191) NOT NULL DEFAULT '';
`
This command will corrupt the database, but the command 
`
ALTER TABLE `applicationconfig` ADD COLUMN     `slugName` VARCHAR(191) NOT NULL DEFAULT '';
`
Works without problem and lets the migration succeed. Again this only happens with the migrate dev command, deploy to prod still works fine.

**Comment by storageddd** [comment]
Yep, solution with lowercase in models works. But I think its not hard to fix it.
Guys, please check it! @janpio

**Comment by storageddd** [comment]
Now I found solution. Please check https://github.com/prisma/prisma/issues/14915

**Comment by Druue** [comment]
@Corydeeee Hi, can you please confirm what container you're running and do you still have the original database configuration?

**Comment by thecoolestcoderrushil** [comment]
A migration that fails to apply to the shadow database is a painful blocker, since the migration can never be resolved even after removing the column. The MySQL table-name capitalization issue causing this is hard to spot. Rehearsing the migration on a production-shaped twin first would surface the shadow-database problem before deploy day. antifailure.dev (https://antifailure.dev) spins up disposable database twins for migration rehearsal, and it is open source and free to try. Happy to set up a demo if that would help.
